web/src/lib/utils/timeline-util.ts

```diff
@@ -45,8 +45,8 @@ export function splitBucketIntoDateGroups(
   assets: AssetResponseDto[],
   locale: string | undefined,
 ): AssetResponseDto[][] {
-  return chain(assets)
-    .groupBy((asset) => fromLocalDateTime(asset.localDateTime).toLocaleString(groupDateFormat, { locale }))
-    .sortBy((group) => assets.indexOf(group[0]))
-    .value();
+  const grouped = groupBy(assets, (asset) =>
+    fromLocalDateTime(asset.localDateTime).toLocaleString(groupDateFormat, { locale }),
+  );
+  return sortBy(grouped, (group) => assets.indexOf(group[0]));
```

Comment:
Unrelated to this PR but isn't this `assets.indexOf` *really* slow given how often we call it?

Reply:
It does look slow 😅